src/main/cpp/src/parse_uri.cu

```diff
@@ -164,11 +166,11 @@ bool __device__ validate_ipv6(string_view s)
   int address_char_count{0};
   bool address_has_hex{false};
 
-  auto const leading_double_colon = [&]() {
+  auto const leading_double_colon = cuda::proclaim_return_type<bool>([&]() {
```

Comment:
This is not device lambda so don't need to change here.

Reply:
Wait, I just realized that this lambda is unused. The only place that refers to it is https://github.com/NVIDIA/spark-rapids-jni/blob/67845706969fe7c9495e2578db7de278729841fe/src/main/cpp/src/parse_uri.cu#L212 which is commented out.

Reply:
Yeah. I wrote it all to match the RFC, but Spark/Java don't follow it exactly and I need to match them. I documented the differences by commenting out the original path. I don't know the take on this and if I should remove it or not.

Reply:
If you commented out the caller then you should also comment out the function too, since it is now unused.